jobbergate-api/jobbergate_api/s3_manager.py

```diff
@@ -40,32 +160,60 @@ def _get_key(self, app_id):
             )
         return self.key_template.format(app_id=app_id)
 
-    def put(self, upload_file: UploadFile, app_id: str = ""):
+    def put(self, upload_file: UploadFile, app_id: str = "") -> str:
         """
-        Upload a file to s3 for the given app_id and returns the.
+        Upload a file to s3 for the given app_id and returns its key.
         """
         key = self._get_key(app_id)
-        self.s3_client.put_object(
-            Body=upload_file.file,
-            Bucket=self.bucket_name,
-            Key=key,
-        )
+        logger.debug(f"Uploading to S3: {key})")
+        if isinstance(upload_file, str):
+            self.client.put(upload_file, key)
+        else:
+            self.client.put(upload_file.file, key)
         return key
 
     def delete(self, app_id: str = ""):
         """
         Delete a file from s3 associated to the given app_id.
         """
         key = self._get_key(app_id)
-        try:
-            self.s3_client.delete_object(Bucket=self.bucket_name, Key=key)
-        except self.s3_client.exceptions.NoSuchBucket:
-            raise KeyError(f"No such bucket: {self.bucket_name}")
-        except self.s3_client.exceptions.NoSuchKey:
-            raise KeyError(f"No such key: {key}")
+        logger.debug(f"Deleting from S3: {key})")
+        self.client.delete(key=key)
 
     def get(self, app_id: str = ""):
         """
         Get a file from s3 associated to the given app_id.
         """
-        return self.s3_client.get_object(Bucket=self.bucket_name, Key=self._get_key(app_id))
+        key = self._get_key(app_id)
+        logger.debug(f"Getting from S3: {key})")
+        return self.client.get(key=key)
+
+    def get_s3_object_as_string(self, app_id) -> str:
+        s3_file_obj = self.get(app_id)
+        if isinstance(s3_file_obj, str):
+            return s3_file_obj
```

Comment:
This block may be unnecessary if we use the S3 fixture properly. Investigate it.